### PR TITLE
Add style guide for alert naming, labels, annotations

### PR DESCRIPTION
**Pull request by beorn7**
This is obviously very opinionated as I just put down what I'm evangelizing myself. But I wanted to get started with a suggestion. Let's take it from here. @brancz @paulfantom

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,3 +151,58 @@ $ jsonnet -J vendor -S -e 'std.manifestYamlDoc((import "mixin.libsonnet").promet
...
+  notification. A notification template can iterate through all the
+  descriptions and format them into a list. Examples (again corresponding to
+  the examples above): `Filesystem on {{ $labels.device }} at {{
+  $labels.instance }} has only {{ printf "%.2f" $value }}% available inodes
+  left.`, `Alert notification queue of Prometheus %(prometheusName)s is running
+  full.`.
+- `runbook_url` (optional): A link to a runbook for the alert. (TODO: Flesh out
+  details. Also see [kubernetes-mixin issue
+  #312](https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/312).)
+- `dashboard_url` (optional): A link to a meaningful Grafana dashboard. (TODO:
```
Comment (paulfantom) [inline README.md]: I am in favor of this annotation field, but it greatly complicates a generic usage of mixins. This is the only field that always needs to be configured by end-user, which is not the case for any other mixin.

Additionally, it does not show up in any of the currently used ones. I would prefer to do a spike on at least one mixin before making this into a general guideline.

File: README.md [hunk 2]
```diff
@@ -151,3 +151,58 @@ $ jsonnet -J vendor -S -e 'std.manifestYamlDoc((import "mixin.libsonnet").promet
...
+- `description` (mandatory): A detailed description of a single alert, with
+  most of the important information templated in. The description usually
+  expands into a different string for every individual alert within a
+  notification. A notification template can iterate through all the
+  descriptions and format them into a list. Examples (again corresponding to
+  the examples above): `Filesystem on {{ $labels.device }} at {{
+  $labels.instance }} has only {{ printf "%.2f" $value }}% available inodes
+  left.`, `Alert notification queue of Prometheus %(prometheusName)s is running
+  full.`.
+- `runbook_url` (optional): A link to a runbook for the alert. (TODO: Flesh out
```
Comment (brancz) [inline README.md]: I like runbook url and dashboard url, but as you already noted, these haven't really been fleshed out enough, I think we should skip them for now

**Review by brancz — COMMENTED** [review]
Happy with moving forward with this modulo runbook and dashboard url

**Comment by brancz** [comment]
cc @metalmatze

**Review by paulfantom — APPROVED** [review]
Awesome doc! Apart from `dashboards_url` I am in favor of such guidelines.

**Comment by beorn7** [comment]
Sure, let's remove `dashboard_url` and `runbook_url`. But perhaps add a note that we intend to add a guideline for those once we have fleshed out the details? That would help users that like those annotation and might assume we haven't even thought about those…

**Comment by brancz** [comment]
Yeah I'm happy with that.

**Review by brancz — APPROVED** [review]

**Comment by metalmatze** [comment]
Amazing stuff!
Sorry for lag, still catching up from everything after 2 offline weeks.

**Pull request merged**